src/dsp_tools/commands/xmlupload/make_rdf_graph/make_values.py

```diff
@@ -51,157 +69,140 @@ def make_values(properties: list[XMLProperty], res_bnode: BNode, lookups: Lookup
+            literal_info = RDF_LITERAL_PROP_TYPE_MAPPER[type(val_type)]
+            properties_graph = _make_value_graph_with_literal_object(
+                val=val,
                 res_bn=res_bnode,
-                prop_name=prop_name,
                 prop_type_info=literal_info,
-                transformer=transformer,
-                permissions_lookup=lookups.permissions,
-            )
-        case "list":
-            properties_graph = _make_list_prop_graph(
-                prop=prop,
-                res_bn=res_bnode,
-                prop_name=prop_name,
-                permissions_lookup=lookups.permissions,
-                listnode_lookup=lookups.listnodes,
             )
-        case "resptr":
-            prop_name = get_absolute_iri(f"{prop.name}Value", lookups.namespaces)
-            properties_graph = _make_link_prop_graph(
-                prop=prop,
+        case IntermediaryList():
+            properties_graph = _make_list_value_graph(val=val, res_bn=res_bnode, prop_type_info=LIST_PROP_TYPE_INFO)
+        case IntermediaryLink():
+            properties_graph = _make_link_value_graph(
+                val=val,
                 res_bn=res_bnode,
-                prop_name=prop_name,
-                permissions_lookup=lookups.permissions,
-                iri_resolver=lookups.id_to_iri,
+                prop_type_info=LINK_PROP_TYPE_INFO,
+                iri_resolver=iri_lookup.id_to_iri,
             )
-        case "text":
-            properties_graph = _make_text_prop_graph(
-                prop=prop,
+        case IntermediaryRichtext():
+            properties_graph = _make_richtext_value_graph(
+                val=val,
                 res_bn=res_bnode,
-                prop_name=prop_name,
-                permissions_lookup=lookups.permissions,
-                iri_resolver=lookups.id_to_iri,
+                prop_type_info=RICHTEXT_PROP_TYPE_INFO,
+                iri_resolver=iri_lookup.id_to_iri,
             )
-        case "date":
-            properties_graph = _make_date_prop_graph(
-                prop=prop,
+        case IntermediaryDate():
+            properties_graph = _make_date_value_graph(
+                val=val,
                 res_bn=res_bnode,
-                prop_name=prop_name,
-                permissions_lookup=lookups.permissions,
             )
-        case "interval":
-            properties_graph = _make_interval_prop_graph(
-                prop=prop,
+        case IntermediaryInterval():
+            properties_graph = _make_interval_value_graph(
+                val=val,
                 res_bn=res_bnode,
-                prop_name=prop_name,
-                permissions_lookup=lookups.permissions,
             )
         case _:
-            raise UserError(f"Unknown value type: {prop.valtype}")
-    return properties_graph, prop_name
+            raise UserError(f"Unknown value type: {type(val).__name__}")
+    return properties_graph, URIRef(val.prop_iri)
```

Comment:
what is the meaning of URIRef(val.prop_iri)? Could you quickly document (perhaps in a function docstring) what this is and what it is used for?